packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,1569 @@
+    if (props.vpcOptions != null && zoneAwarenessEnabled &&
+      new Set(props.vpcOptions?.subnets.map((subnet) => subnet.availabilityZone)).size < availabilityZoneCount) {
+      throw new Error('When providing vpc options you need to provide a subnet for each AZ you are using');
+    };
+
+    if ([dedicatedMasterType, instanceType].some(t => !t.endsWith('.elasticsearch'))) {
+      throw new Error('Master and data node instance types must end with ".elasticsearch".');
+    }
+
+    const elasticsearchVersion = props.version.version;
+    const elasticsearchVersionNum = parseVersion(props.version);
+
+    if (
+      elasticsearchVersionNum <= 7.7 &&
+      ![
+        1.5, 2.3, 5.1, 5.3, 5.5, 5.6, 6.0,
+        6.2, 6.3, 6.4, 6.5, 6.7, 6.8, 7.1, 7.4,
+        7.7,
+      ].includes(elasticsearchVersionNum)
+    ) {
+      throw new Error(`Unknown Elasticsearch version: ${elasticsearchVersion}`);
+    }
+
+    const unsignedBasicAuthEnabled = props.useUnsignedBasicAuth ?? false;
+
+    if (unsignedBasicAuthEnabled) {
+      if (props.enforceHttps == false) {
+        throw new Error('You cannot disable HTTPS and use unsigned basic auth');
+      }
+      if (props.nodeToNodeEncryption == false) {
+        throw new Error('You cannot disable node to node encryption and use unsigned basic auth');
+      }
+      if (props.encryptionAtRest?.enabled == false) {
+        throw new Error('You cannot disable encryption at rest and use unsigned basic auth');
+      }
+    }
+
+    const unsignedAccessPolicy = new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: ['es:ESHttp*'],
+      principals: [new iam.Anyone()],
+      resources: [cdk.Lazy.stringValue({ produce: () => `${this.domainArn}/*` })],
+    });
+
+    const masterUserArn = props.fineGrainedAccessControl?.masterUserArn;
+    const masterUserNameProps = props.fineGrainedAccessControl?.masterUserName;
+    // If basic auth is enabled set the user name to admin if no other user info is supplied.
+    const masterUserName = unsignedBasicAuthEnabled
+      ? (masterUserArn == null ? (masterUserNameProps ?? 'admin') : undefined)
+      : masterUserNameProps;
+
+    if (masterUserArn != null && masterUserName != null) {
+      throw new Error('Invalid fine grained access control settings. Only provide one of master user ARN or master user name. Not both.');
+    }
+
+    const advancedSecurityEnabled = (masterUserArn ?? masterUserName) != null;
+    const internalUserDatabaseEnabled = masterUserName != null;
+    const masterUserPasswordProp = props.fineGrainedAccessControl?.masterUserPassword;
+    const createMasterUserPassword = (): cdk.SecretValue => {
+      return new secretsmanager.Secret(this, 'MasterUser', {
+        generateSecretString: {
+          secretStringTemplate: JSON.stringify({
+            username: masterUserName,
+          }),
+          generateStringKey: 'password',
```

Comment:
There are a few characters that make it a real pain to put in terminals (requires fancy escaping), so suggestion here is just to get rid of them in the generation.

```suggestion
          generateStringKey: 'password',
          excludeCharacters: "{}'\\*[]()`",
```